src/Logary/Registry.fs

```diff
@@ -144,51 +137,55 @@ module Advanced =
   /// how that went. E.g. if there's a target is has a backlog, it might not be able to
   /// flush all its pending entries within the allotted timeout (200 ms at the time of writing)
   /// but will then instead return Nack/failed RPC.
-  let flushPending dur (registry : IActor) =
-    if registry.Status <> ActorStatus.Running then async { return Nack "registry not running" }
-    /// Timeout=Infinite below is for THIS call, internally registry has a timeout per target.
-    else registry |> Actor.reqReply (fun c -> FlushPending(dur, c)) Infinite
+  let flushPending (registry : RegistryInstance) : Alt<unit> = Alt.withNackJob <| fun nack -> job {
+    let! flushCh = Ch.create ()
+    do! registry.reqCh *<+ (FlushPending (flushCh, nack))
+    return flushCh
+  }
```

Comment:
While the pattern using withNack above is correct, on deeper look, it would seem that for this operation nack is unnecessary.  It would seem that this operation (flushing) is essentially an idempotent operation.  Therefore you should be able to do with just `prepare`.  Something like this:

```fsharp
let flushPending (registry : RegistryInstance) : Alt<unit> =
  Alt.prepareJob <| fun () ->
  let flushIv = IVar ()
  registry.reqCh *<+ FlushPending flushIv >>-
  flushIv
```